test_complete_functionality.py

```
@@ -0,0 +1,230 @@
+        if ready[0]:
+            response = process.stdout.readline()
+            if response:
+                data = json.loads(response)
+                if "result" in data:
+                    print_test("MCP protocol initialization", "PASS")
+                    process.terminate()
+                    return True
+
+        process.terminate()
+        print_test("MCP protocol initialization timeout", "WARN")
+        return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)